numba/cpython/setobj.py

```diff
@@ -1188,10 +1246,12 @@ def set_constructor(context, builder, sig, args):
 
     # If the argument has a len(), preallocate the set so as to
     # avoid resizes.
+    # `for_iter` increfs each item in the set, so don't incref in add
```

Comment:
I don't think it would ever be used, because here the decref is necessary. This is because the constructor could be a list of repeating items, in which case we would want the number of increfs to be 1 because the set only contains the item once. Therefore, the `decref` is needed to balance the `incref` from `for_iter`, and the incref from `.add` will only occur the one time a repeated item is added to the set and skipped every other time because the item is already in the set

Reply:
Thanks for the explanation, this makes sense. Please could the comment be updated with/to reflect this explanation, many thanks!